ReactNativeClient/lib/joplin-renderer/MdToHtml/rules/embedvideo.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+// The youtubeParser is from http://stackoverflow.com/a/8260383
+function youtubeParser(url) {
+  var regExp = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#&?]*).*/;
+  var match = url.match(regExp);
+  if (match && match[7].length == 11) {
+    return match[7];
+  } else {
+    return url;
+  }
+}
+
+function videoEmbed(md) {
+  function videoReturn(state, silent) {
+    var serviceEnd,
+      serviceStart,
+      token,
+      videoID,
+      theState = state;
+    const oldPos = state.pos;
+
+    var extractURL = /\[(youtube)\]\([\s]*(.*?)[\s]*[)]/im;
+
+    if (state.src.charCodeAt(state.pos) !== 0x5b /* [ */) {
+      return false;
+    }
```

Comment:
That solution is not optimum as you're unnecessarily parsing the whole Markdown document even though it's already been parsed by markdown-it. Markdown-it will produce a stream of tokens, which you can hook into to provide alternative rendering. In your case, you could hook in link_open and, if you detect that the href is a YouTube URL, display your iframe code. Please have a look at the markdown-it doc for more info.

Reply:
I'll look into the same!